Core/Core/Views/HorizontalMenuView.swift

```diff
@@ -0,0 +1,132 @@
+// Copyright © 2019 Instructure, Inc. All rights reserved.
+
+import UIKit
+
+public protocol HorizontalMenuDelegate: class {
+    var maxItemWidth: CGFloat { get }
+    var measurementFont: UIFont { get }
+    var selectedColor: UIColor? { get }
+    func menuItemCount() -> Int
+    func menuItemTitle(at: IndexPath) -> String
+    func didSelectItem(at: IndexPath)
+}
+
+public class HorizontalMenuView: UIView {
+    @IBOutlet private weak var contentView: UIView!
+    @IBOutlet private weak var collectionView: UICollectionView!
+    private var selectedIndexPath: IndexPath? // = IndexPath(item: 0, section: 0)
+    public weak var delegate: HorizontalMenuDelegate?
+    private var cachedTotalWidthOfItems: CGFloat?
+    private var cachedSpaceBetweenCells: CGFloat?
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+
+    func commonInit() {
+        Bundle.core.loadNibNamed("HorizontalMenuView", owner: self, options: nil)
```

Comment:
`loadFromXib()` does a bunch of this for you already, doesn't it?

Reply:
I'm guessing I did this because `loadFromXIB` defaults to the main bundle.  This is all coming from the `Core` bundle.

Reply:
`loadFromXib` is our custom extension that loads from whatever bundle the class lives in. it then adds the top level view as a subview and pins it inside.

Reply:
i can take another look at it

Reply:
It looks like you tried to use the static loadFromXib, there is also an instance loadFromXib, which seems more like what you are doing.